blocklylib-core/src/main/java/com/google/blockly/android/codegen/LazyServiceConnection.java

```diff
@@ -0,0 +1,71 @@
+package com.google.blockly.android.codegen;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+
+import com.google.blockly.android.codegen.CodeGeneratorService.CodeGeneratorBinder;
+
+class LazyServiceConnection {
+
+    private final Context mContext;
+    private ServiceConnection mCodeGenerationConnection;
+
+    private CodeGeneratorService mGeneratorService;
+    private CodeGenerationRequest mStoredRequest;
+
+    LazyServiceConnection(Context context) {
+        this.mContext = context;
+    }
+
+    void requestCodeGeneration(CodeGenerationRequest codeGenerationRequest) {
+        if (isBound()) {
+            executeCodeGenerationRequest(codeGenerationRequest);
+        } else {
+            mStoredRequest = codeGenerationRequest;
```

Comment:
I think this is an edge case, but I've added a queue anyway.
If you use different callback objects with different code generation requests then this all works as expected.